andela/ah-the-answer-backend
# #165165013 Prevent non alphanumeric password signup

__What does this PR do?__
Prevents a user from signing up with a password that is not alphanumeric.

__Description of tasks to be completed__

- use this  `"^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$"`. regex that checks if the password comprises of a minimum eight characters, at least one letter and one number:

__How should this be manually tested__

- Clone the repo.

- Create a virtual environment and activate it using the commands below.

       - `virtualenv venv`
       - `source venv/bin/activate`
- Install all the projects requirements using the command below.

     ` - pip install -r requirements.txt`

- Run the local server
      `- python manage.py runserver`

- test the endpoint on postman by navigating to `http://127.0.0.1:8000/api/users/ ` and adding the payload below on the body section of postman then hit send

```
 {
	"user":{
		"email":"example@gmail.com",
		"password":"asdfghjklsdfghj",
		"username":"tesuser"
	}
}
```

you should expect to get an error message as the one below

```
{
    "errors": {
        "password": [
            "Please ensure your password contains at least one letter and one numeral"
        ]
    }
}
```

__Relevant Pivotal tracker stories__
https://www.pivotaltracker.com/story/show/165165013
